wasp-ai/src/client/pages/MainPage.jsx

```diff
@@ -68,12 +70,51 @@ const MainPage = () => {
 
   const [appAuthMethod, setAppAuthMethod] = useState(availableAuthMethods[0]);
 
+  useEffect(() => {
+    try {
+      const appDetails = JSON.parse(localStorage.getItem("appDetails"));
+      const appNum = JSON.parse(localStorage.getItem("appNum"));
+      if (!appNum) {
+        localStorage.setItem("appNum", 0);
+      }
+      if (appNum === 2) {
+        setIsGhModalOpen(true);
+      }
+      if (appDetails) {
+        setAppName(appDetails.appName);
+        setAppDesc(appDetails.appDesc);
+        setAppPrimaryColor(availableColors.find((color) => color.name === appDetails.appPrimaryColor));
+        setAppAuthMethod(availableAuthMethods.find((method) => method.value === appDetails.appAuthMethod));
+        setCreativityLevel(availableCreativityLevels.find((level) => level.value === appDetails.appCreativityLevel));
+        localStorage.removeItem("appDetails");
+      }
+    } catch (error) {
+      console.error(error);
+    }
+  }, []);
+
```

Comment:
I thoiught it would be nice to save their app description and repopulate it on return to the main page, so the prompt and settings are to adjust/modify on regenerations

Reply:
Ah got it, that is pretty cool!

Let's improve the code so the intention here is obvious. You could have two functions, one called "locallySaveLastGenerationInput" and "locallyLoadLastGenerationInput". That way you hide a bit the implementation details, plus the code is more readable and intention is clearer. Maybe there is a bit more that could be done here but it will be more obvious after these changes.